src/ppx/grfx/dx12/dx12_pipeline.cpp

```diff
@@ -289,30 +289,67 @@ Result PipelineInterface::CreateApiObjects(const grfx::PipelineInterfaceCreateIn
         for (size_t bindingIndex = 0; bindingIndex < bindings.size(); ++bindingIndex) {
             const grfx::DescriptorBinding& binding = bindings[bindingIndex];
 
-            // Allocate unique range
-            std::unique_ptr<D3D12_DESCRIPTOR_RANGE1> range = std::make_unique<D3D12_DESCRIPTOR_RANGE1>();
-            if (!range) {
-                PPX_ASSERT_MSG(false, "allocation for descriptor range failed for set=" << set << ", binding=" << binding.binding);
-                return ppx::ERROR_ALLOCATION_FAILED;
+            // If set is pushable then add as root descriptor...
+            if (pLayout->IsPushable()) {
+                // Figure out which root descriptor parameter type
+                D3D12_ROOT_PARAMETER_TYPE parameterType = InvalidValue<D3D12_ROOT_PARAMETER_TYPE>();
+                switch (binding.type) {
+                    default: {
+                        PPX_ASSERT_MSG(false, "Descriptor binding type is not pushable in Direct3D 12");
```

Comment:
This wouldn't work because it would just stop BigWheels from building all together. It has to be a runtime error.

Reply:
Oh, shoot!  You're right.  So, we shouldn't build the DX12 when building a `vk_...` binary then.  That would take more work in the build system.  But, it's doable, isn't it?

Reply:
I think it would still need to be a runtime check. It's not really worth the effort to put such an egregious check in place. If you disable D3D12, it won't include the D3D12 build files. The only platform that it builds D3D12 is on Windows. It would also just get really confusing to debug later should there be a problem.

Reply:
Yeah, we'd need a better separation between the two APIs.  Right now everything is built together.  Perhaps a cleanup for a later time.